test/fuse_pointwise.cpp

```diff
@@ -361,4 +361,75 @@ TEST_CASE(no_input)
     EXPECT(p == p2);
 }
 
+TEST_CASE(add_reshape_add)
+{
+    migraphx::shape s1{migraphx::shape::float_type, {3, 10, 16}};
+    migraphx::shape s2{migraphx::shape::float_type, {3, 40, 2, 2}};
+    migraphx::shape s3{migraphx::shape::float_type, {3, 10, 4, 2, 2}};
+    migraphx::program p1;
+    {
+        auto* mm  = p1.get_main_module();
+        auto x    = mm->add_parameter("x", s1);
+        auto y    = mm->add_parameter("y", s1);
+        auto z    = mm->add_parameter("z", s2);
+        auto add1 = mm->add_instruction(migraphx::make_op("add"), x, y);
+        auto reshape =
+            mm->add_instruction(migraphx::make_op("reshape", {{"dims", s2.lens()}}), add1);
+        auto add2 = mm->add_instruction(migraphx::make_op("add"), reshape, z);
+        mm->add_return({add2});
+    }
+    run_pass(p1);
+    migraphx::program p2;
+    {
+        auto* mm = p2.get_main_module();
+        auto x   = mm->add_parameter("x", s1);
+        auto y   = mm->add_parameter("y", s1);
+        auto z   = mm->add_parameter("z", s2);
+        auto x2  = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), x);
+        auto y2  = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), y);
+        auto z2  = mm->add_instruction(migraphx::make_op("reshape", {{"dims", s3.lens()}}), z);
+        auto fadd =
+            add_pointwise(p2, "main:pointwise0", {x2, y2, z2}, [=](auto* pm, const auto& inputs) {
+                auto add1 = pm->add_instruction(migraphx::make_op("add"), inputs[0], inputs[1]);
+                return pm->add_instruction(migraphx::make_op("add"), add1, inputs[2]);
+            });
+        auto reshape =
+            mm->add_instruction(migraphx::make_op("reshape", {{"dims", s2.lens()}}), fadd);
+        mm->add_return({reshape});
+    }
+    EXPECT(p1.sort() == p2.sort());
+}
+
+TEST_CASE(add_reshape_add_error)
```

Comment:
Why does this fail?

Reply:
Because we cant compute a common dimensionality that preserves the axes. This can cause us to insert extra contiguous. For example if one of the inputs are broadcasted along the first axis of `s1`(ie a shape of `{6, 35}, {1, 0}`) and we reshape to `{3, 7, 2, 5}` then we will need to insert an extra contiguous as there is no way to set the strides to handle this case because `6` is being mapped to `3, 7` which is not the same number of elements.

In general, models do not do reshapes like this, so it shouldn't be a problem in practice.

Reply:
Ah, makes sense. So the test, in some sense - at least how I'd describe it, is that you can only fuse when the reshape only
- Takes multiple contiguous dimensions from the input and merges them, or
- Takes one dimension from the input and splits it evenly

Reply:
> inputs are broadcasted along the first axis of s1(ie a shape of {6, 35}, {1, 0}) and we reshape to {3, 7, 2, 5} 

If input shapes are standard then it should be possible to reshape it to {3, 7, 2, 5} I think.  

That's what it is doing here : 
`auto reshape =
            mm->add_instruction(migraphx::make_op("reshape", {{"dims", s2.lens()}}), add1);`

Reply:
>  and splits it evenly 

it could be uneven split (e.g. 35--> 5 * 7) from what i understand.

Reply:
> If input shapes are standard then it should be possible to reshape it to {3, 7, 2, 5} I think.

Yes of course, but we wont know if the shapes will be standard or not until after eliminate_contiguous. Doing it prematurely will introduce more contiguous operators.

> it could be uneven split (e.g. 35--> 5 * 7) from what i understand.

Reshape doesnt handle such reshapes with non-standard shape. Furthermore, the calculation of the axes in common_dims would be a lot more complicated.